aten/src/ATen/functorch/ADInterpreters.cpp

```diff
@@ -136,12 +156,29 @@ static void autogradBasedTransformSendToNext(
   if (getDynamicLayerStack().size() == 0) {
     sanityCheckStack(op, stack);
   }
-  op.callBoxed(stack);
 
   // Step 4, 5, 6
   auto ret_size = op.schema().returns().size();
+
+  op.callBoxed(stack);
+
+  auto aliased_input_outputs = findAliasedInputs(op.schema());
+  std::vector<int64_t> unwrapped_outputs; // indexes of outputs that should remain unwrapped
+
+  // lift_fresh: it's must be freshly allocated and should be wrapped. User shouldn't have access to input version
+  // alias: this is needed for the CompositeImplicit instance norm (running_mean/var get set to be a wrapped value)
+  //        It's not a user facing function, but is more prone to possible errors
```

Comment:
Why is there a special case for alias?

Reply:
Do you remember what tests failed without this? (I want to step into it a little)

Reply:
This one is actually saved on hud (bless ghstack)

Tests: 
* `test_vjpvjp_nn_functional_instance_norm`
* `test_jvpvjp_nn_functional_instance_norm`

Error:
`RuntimeError: During a grad (vjp, jvp, grad, etc) transform, the function provided attempted to call in-place operation (aten::copy_) that would mutate a captured Tensor. This is not supported; please rewrite the function being transformed to explicitly accept the mutated Tensor(s) as inputs.`
<details>
<summary>Full stacktrace</summary>

```
Traceback (most recent call last):
2022-09-22T00:19:00.5652985Z   File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 391, in instantiated_test
2022-09-22T00:19:00.5653466Z     raise rte
2022-09-22T00:19:00.5654132Z   File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 378, in instantiated_test
2022-09-22T00:19:00.5691739Z     result = test(self, **param_kwargs)
2022-09-22T00:19:00.5692673Z   File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 815, in test_wrapper
2022-09-22T00:19:00.5693207Z     return test(*args, **kwargs)
2022-09-22T00:19:00.5693656Z   File "/var/lib/jenkins/workspace/functorch/test/test_ops.py", line 1218, in test_jvpvjp
2022-09-22T00:19:00.5694195Z     result = jvp(push_vjp, (primals, cotangents), (primals_tangents, cotangents_tangents))
2022-09-22T00:19:00.5694912Z   File "/opt/conda/lib/python3.7/site-packages/functorch/_src/eager_transforms.py", line 801, in jvp
2022-09-22T00:19:00.5695519Z     return _jvp_with_argnums(func, primals, tangents, argnums=None, strict=strict, has_aux=has_aux)
2022-09-22T00:19:00.5696288Z   File "/opt/conda/lib/python3.7/site-packages/functorch/_src/eager_transforms.py", line 850, in _jvp_with_argnums
2022-09-22T00:19:00.5696741Z     result_duals = func(*duals)
2022-09-22T00:19:00.5697174Z   File "/var/lib/jenkins/workspace/functorch/test/test_ops.py", line 1215, in push_vjp
2022-09-22T00:19:00.5697592Z     _, vjp_fn = vjp(fn, *primals)
2022-09-22T00:19:00.5698184Z   File "/opt/conda/lib/python3.7/site-packages/functorch/_src/eager_transforms.py", line 273, in vjp
2022-09-22T00:19:00.5698699Z     return _vjp_with_argnums(func, *primals, has_aux=has_aux)
2022-09-22T00:19:00.5699402Z   File "/opt/conda/lib/python3.7/site-packages/functorch/_src/eager_transforms.py", line 300, in _vjp_with_argnums
2022-09-22T00:19:00.5699984Z     primals_out = func(*primals)
2022-09-22T00:19:00.5700405Z   File "/var/lib/jenkins/workspace/functorch/test/test_ops.py", line 118, in wrapped
2022-09-22T00:19:00.5701065Z     result = f(*_args, **kwargs)
2022-09-22T00:19:00.5701762Z   File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/opinfo/core.py", line 1026, in __call__
2022-09-22T00:19:00.5702217Z     return self.op(*args, **kwargs)
2022-09-22T00:19:00.5702830Z   File "/opt/conda/lib/python3.7/site-packages/torch/nn/functional.py", line 2492, in instance_norm
2022-09-22T00:19:00.5703459Z     input, weight, bias, running_mean, running_var, use_input_stats, momentum, eps, torch.backends.cudnn.enabled
2022-09-22T00:19:00.5704931Z RuntimeError: During a grad (vjp, jvp, grad, etc) transform, the function provided attempted to call in-place operation (aten::copy_) that would mutate a captured Tensor. This is not supported; please rewrite the function being transformed to explicitly accept the mutated Tensor(s) as inputs.
```

</details>

https://hud.pytorch.org/pytorch/pytorch/pull/85374?sha=9c002fb612fa90a879280181c59df16fee9f81ca

Reply:
Also I'm pretty sure the copy_ is coming from here
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Normalization.cpp#L633